File: android/app/src/main/java/wilton/calls/fs/WriteFile.java
/*
 * Copyright 2020, alex at staticlibs.net
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package wilton.calls.fs;

import java.io.File;

import wilton.Call;

import static org.apache.commons.codec.binary.Hex.decodeHex;
import static org.apache.commons.io.FileUtils.writeByteArrayToFile;
import static org.apache.commons.io.FileUtils.writeStringToFile;
import static mobile.example.utils.JsonUtils.GSON;

public class WriteFile implements Call {
    @Override
    public String call(String data) throws Exception {
        Options opts = GSON.fromJson(data, Options.class);
        File file = new File(opts.path);
        if (opts.hex) {
            byte[] bytes = decodeHex(data.toCharArray());
            writeByteArrayToFile(file, bytes);
        } else {
            writeStringToFile(file, opts.data, "UTF-8");
        }
        return null;
    }

    private static class Options {
        String path;
        String data;
        boolean hex = false;
    }
}

File: ios/ios/Wilton/Calls/Fs/WriteFile.swift
//
//  WriteFile.swift
//  ios
//
//  Created by Alexey Liverty on 6/7/20.
//  Copyright © 2020 alex. All rights reserved.
//

import Foundation

class WriteFile : Call {
    func call(_ data: String) throws -> String {
        let opts = try wiltonFromJson(data, Options.self)
        guard let path = opts.path else {
            throw WiltonException("Fs/WriteFile: Required parameter 'path' not specified")
        }
        guard let data = opts.data else {
            throw WiltonException("Fs/WriteFile: Required parameter 'data' not specified")
        }
        do {
            if opts.hex {
                let bytes = decodeHex(data)
                try writeBytesToFile(path, bytes)
            } else {
                try writeStringToFile(path, data)
            }
        return ""
        } catch {
            throw WiltonException("Fs/ReadDir: Error writing file, path: [\(path)], message: [\(error)]")
        }
    }
    
    class Options : Codable {
        var path: String?
        var data: String?
        var hex: Bool = false
    }
}
